### PR TITLE
AppArmor: make policy field optional if abstract policy is specified

### PR DESCRIPTION
**Pull request by mhils**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

This PR makes the `policy` field optional for AppArmor policies.

#### Which issue(s) this PR fixes:

#2388

#### Does this PR have test?

Covered by existing tests

#### Does this PR introduce a user-facing change?

```release-note
AppArmor profiles can now have either an abstract or a concrete policy.
```

### CODE REVIEW

File: internal/pkg/daemon/apparmorprofile/apparmor_supported.go [hunk 1]
```diff
@@ -74,7 +76,23 @@ func (a *aaProfileManager) InstallProfile(bp profilebasev1alpha1.StatusBaseUser)
...
+		policy, err = crd2armor.GenerateProfile(profile.GetProfileName(), &profile.Spec.Abstract)
+		if err != nil {
+			return false, fmt.Errorf("generating raw apparmor profile: %w", err)
+		}
+	}
+	if profile.Spec.Policy != "" {
+		if policy != "" && policy != profile.Spec.Policy {
+			return false, errors.New("abstract and concrete policy do not match")
+		}
+		policy = profile.Spec.Policy
```
[the remark this answers is not in the document]
Comment (mhils) [reply internal/pkg/daemon/apparmorprofile/apparmor_supported.go:94]: I think the previous logic was correct (the assignment was required as the check only checked for mismatches), but this is still good feedback that things weren't easy to follow here. I've updated the logic to be more easy to follow, PTAL. :)

**Comment by ccojocar** [comment]
I think you need to regenerate the bundle since the profile manifest was changed.

**Comment by saschagrunert** [comment]
Please rebase this one as well.

**Comment by saschagrunert** [comment]
The OLM bundles do not seem to be up to date: https://github.com/kubernetes-sigs/security-profiles-operator/actions/runs/11300455856/job/31483016900?pr=2469

**Comment by mhils** [comment]
Thanks @saschagrunert! Is it intentional that `make verify` does not include `make verify-bundle`? 😅

**Comment by saschagrunert** [comment]
> Thanks @saschagrunert! Is it intentional that `make verify` does not include `make verify-bundle`? 😅

Hm, we could add it indeed. :thinking:

**Review by saschagrunert — APPROVED** [review]
/ok-to-test

**Pull request merged**